iflist.go

```diff
@@ -0,0 +1,143 @@
+package nmap
+
+import (
+	"bytes"
+	"net"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// InterfaceList contains interfaces and routes
+type InterfaceList struct {
+	Interfaces []*Interface `json:"interfaces"`
+	Routes     []*Route     `json:"routes"`
+}
+
+// Interface is a interface object
+type Interface struct {
+	Device string           `json:"device"`
+	Short  string           `json:"short"`
+	IP     net.IP           `json:"ip"`
+	IPMask net.IP           `json:"ip_mask"`
+	Type   string           `json:"type"`
+	Up     bool             `json:"up"`
+	MTU    int              `json:"mtu"`
+	Mac    net.HardwareAddr `json:"mac"`
+}
+
+// Route is a route object
+type Route struct {
+	DestinationIP     net.IP `json:"destination_ip"`
+	DestinationIPMask net.IP `json:"destination_ip_mask"`
+	Device            string `json:"device"`
+	Metric            int    `json:"metric"`
+	Gateway           net.IP `json:"gateway"`
+}
+
+// GetInterfaceList runs nmap with --iflist option. The output will be parsed.
+// The return value is a struct containing all host interfaces and routes.
+func (s *Scanner) GetInterfaceList() (result *InterfaceList, err error) {
+	var stdout, stderr bytes.Buffer
+
+	args := append(s.args, "--iflist")
+
+	// Prepare nmap process
+	cmd := exec.Command(s.binaryPath, args...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	// Run nmap process
+	err = cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	result = parseInterfaces(stdout.Bytes())
+
+	return result, nil
+}
+
+func parseInterfaces(content []byte) *InterfaceList {
+	list := InterfaceList{
+		Interfaces: make([]*Interface, 0),
+		Routes:     make([]*Route, 0),
+	}
+	output := string(content)
+	lines := strings.Split(output, "\n")
+
+	for i, line := range lines {
+		if match, _ := regexp.MatchString(`[\*]INTERFACES[\*]`, line); match {
+			for _, line := range lines[i+2:] {
+				if iface := convertInterface(line); iface != nil {
+					list.Interfaces = append(list.Interfaces, iface)
+				}
+			}
+		}
+
+		if match, _ := regexp.MatchString(`[\*]ROUTES[\*]`, line); match {
+			for _, line := range lines[i+2:] {
+				if route := convertRoute(line); route != nil {
+					list.Routes = append(list.Routes, route)
+				}
+			}
+		}
+	}
+
+	return &list
+}
+
+func convertInterface(line string) *Interface {
+	splitted := strings.Fields(line)
```

Comment:
Split is an irregular verb and it's past tense form is also `split`, so I'd prefer if the variable was just called `split`.

In this case though, I think an even better name would be `fields`. WDYT?

Reply:
You're so right :see_no_evil: